### DISCUSSION TITLE
Is it possible to turn off the automatic adjustment function of text elements?

### DISCUSSION THREAD

**Discussion by goddade** [ImpulseAdventure/GUIslice#554] [Q&A] [answered]
Like a button, changing the text does not change the size.

**Comment by Pconti31** [comment]
@goddade Could give a quick example of what you want to implement.
Paul--

**Reply by goddade** [reply]
A text element as big as the screen, I want a text element not to resize when the font or size of the content changes, easy to keep the text in the center of the screen.

**Comment by Pconti31** [comment]
@goddade Are you talking about GUIslice API or the GUIslice Builder?

**Reply by goddade** [reply]
They have the same behavior, right?

**Comment by Pconti31** [comment, marked as answer]
@goddade I may not really understand what the issue is but since you started with a button I'll take a guess,
Buttons keep text centered by setting the Text Alignment=GSLC_ALIGN_MID_MID
You can do the same with text.

![text](https://github.com/user-attachments/assets/5efb8e61-aa01-47b3-b5fb-e863b2ef5b15)

Even when you change content of pElemTxtOut1 to something different it will automatically be centered.
Say like:
```
gslc_ElemSetTxtStr(&m_gui, m_pElemTxtOut1, "new content");
```

Paul--

**Reply by goddade** [reply]
Thank you, I understand.
